core/src/main/scala/chisel3/BlackBox.scala

```diff
@@ -17,6 +18,18 @@ package internal {
     // is not possible to properly name EmptyExtModule created from
     // Defintions. See unit test SeparateElaborationSpec #4.a
     private[chisel3] def _isImportedDefinition: Boolean = false
+
+    /** Return the layes which are supported by this `BlackBox`.
+      *
+      * E.g., if this was a [[BlackBox]] that points at Verilog built from
+      * _another_ Chisel elaboration, then this would be the layers that were
+      * defined in that circuit.
+      *
+      * @note This will cause the emitted FIRRTL to include the `knownlayer`
+      * keyword on the `extmodule` declaration.
+      */
+    def knownLayers: Seq[Layer]
+    knownLayers.foreach(layer.addLayer)
```

Comment:
If the user implements knownLayers with a `val` you will get a `NullPointerException`, this should use `atModuleBodyEnd` or otherwise be in the Module "teardown" logic, i.e. in `generateComponent()`.